plugin.go

```diff
@@ -0,0 +1,33 @@
+//  Copyright (c) 2020 Couchbase, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 		http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package zap
+
+import (
+	"github.com/blevesearch/bleve/index/scorch/segment"
+)
+
+type ZapPlugin struct{}
```

Comment:
Many a times while reading code, I find it difficult to track/load into mind about 
-whether the current code is an implementation of any interface Or if so, where can I find the interface details.. 
Not sure of the right of way of doing this.. But having a comment here referring to the segment Plugin interface would be desirable. Also understand that the plugin code isn't there yet in blevesearch/bleve..
Some way to easily trace back the connection bw interface & implementation is nice..

Reply:
Tried to address in second commit.